crates/topos/src/components/tce/services/push_certificate.rs

```diff
@@ -203,3 +224,68 @@ impl Parser<NodeList> for InputFormat {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::check_certificate_delivery;
+    use rstest::*;
+    use std::time::Duration;
+    use topos_core::api::grpc::tce::v1::StatusRequest;
+    use topos_test_sdk::tce::create_network;
+    use tracing::{debug, info};
+
+    #[rstest]
+    #[test_log::test(tokio::test)]
+    #[timeout(Duration::from_secs(120))]
```

Comment:
```suggestion
    #[timeout(Duration::from_secs(10))]
```

120sec is a little too much I think ^^

Reply:
10 seconds will timeout always